pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -94,25 +96,82 @@ func (p *podAffinityPriorityMap) processTerms(terms []v1.WeightedPodAffinityTerm
+// CalculateInterPodAffinityPriorityMap calculate the number of matching pods on the passed-in "node",
+// and return the number as Score.
+func (ipa *InterPodAffinity) CalculateInterPodAffinityPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *schedulernodeinfo.NodeInfo) (framework.NodeScore, error) {
+	node := nodeInfo.Node()
+	if node == nil {
+		return framework.NodeScore{}, fmt.Errorf("node not found")
+	}
+
+	var topologyScore topologyPairToScore
+	if priorityMeta, ok := meta.(*priorityMetadata); ok {
+		topologyScore = priorityMeta.topologyScore
+	}
+	if topologyScore == nil {
+		return framework.NodeScore{}, nil
+	}
+
+	var score int64
+	for tpKey, tpValues := range topologyScore {
+		if v, exist := node.Labels[tpKey]; exist {
+			score += tpValues[v]
+		}
+	}
+
+	return framework.NodeScore{Name: node.Name, Score: score}, nil
+}
+
+// CalculateInterPodAffinityPriorityReduce normalizes the score for each filteredNode,
+// The basic rule is: the bigger the score(matching number of pods) is, the smaller the
+// final normalized score will be.
+func (ipa *InterPodAffinity) CalculateInterPodAffinityPriorityReduce(pod *v1.Pod, meta interface{}, sharedLister schedulerlisters.SharedLister,
+	result framework.NodeScoreList) error {
+	var topologyScore topologyPairToScore
+	if priorityMeta, ok := meta.(*priorityMetadata); ok {
+		topologyScore = priorityMeta.topologyScore
+	}
+	if topologyScore == nil {
+		return nil
+	}
+
+	var maxCount, minCount int64
+	for i := range result {
+		score := result[i].Score
+		if score > maxCount {
+			maxCount = score
+		}
+		if score < minCount {
+			minCount = score
+		}
+	}
+
+	maxMinDiff := maxCount - minCount
+	for i := range result {
+		fScore := float64(0)
+		if maxMinDiff > 0 {
+			fScore = float64(framework.MaxNodeScore) * (float64(result[i].Score-minCount) / float64(maxCount-minCount))
+		}
+
+		result[i].Score = int64(fScore)
+	}
+
+	return nil
+}
+
+func buildTopologyPairToScore(
+	pod *v1.Pod,
+	sharedLister schedulerlisters.SharedLister,
+	filteredNodes []*v1.Node,
+	allNodes []*schedulernodeinfo.NodeInfo,
+	hardPodAffinityWeight int32,
+) topologyPairToScore {
 	affinity := pod.Spec.Affinity
 	hasAffinityConstraints := affinity != nil && affinity.PodAffinity != nil
 	hasAntiAffinityConstraints := affinity != nil && affinity.PodAntiAffinity != nil
 
 	// pm stores (1) all nodes that should be considered and (2) the so-far computed score for each node.
-	pm := newPodAffinityPriorityMap(nodes)
-	allNodes, err := sharedLister.NodeInfos().List()
-	if err != nil {
-		return nil, err
-	}
-
-	// convert the topology key based weights to the node name based weights
-	var maxCount, minCount int64
+	pm := newPodAffinityPriorityMap(filteredNodes)
 
 	processPod := func(existingPod *v1.Pod) error {
 		existingPodNodeInfo, err := sharedLister.NodeInfos().Get(existingPod.Spec.NodeName)
```

Comment:
A nit: we can pass `Node` to processPod, so don't need to rely on `sharedLister`, hence the method `buildTopologyPairToScore` can eliminate the parameter `sharedLister`.

Reply:
After #84824 merged, we could keep the sharedLister and remove allNodes from the arguments lists. :)